[RNMobile] File block V - Making Download button use RichText (Take 2)

etoledom opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
<!-- Please describe what you have changed or added -->

This PR is an alternative to https://github.com/WordPress/gutenberg/pull/27201.
Patching the RichText size issue using the same strategy used by the Button block.

This approach works for both platforms 🎉 

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

- Add a File block
- Check that the Download button layout is correct.
- Test change the button's text and removing all text to show the placeholder.
  - **Note:** The button must be deselected for the Placeholder to show.


## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

etoledom committed:
Download button as RichText

etoledom committed:
Update unit tests

jd-alexander reviewed (commented):
Thanks for these changes @etoledom the Download button is behaving as expected. I am seeing this strange issue on Android when typing a long label for the button. Did you notice this? I am going to see if this is an emulator issue and investigate what could be causing it. 
<kbd><img src="https://user-images.githubusercontent.com/1509205/100310842-2c437280-2f7c-11eb-9c47-c43b4c0d6203.gif" width="320"></kbd>

etoledom committed:
Fix merge errors

etoledom committed:
Fix android download button with long text

etoledom committed:
Update unit test snapshot

etoledom commented:
Thanks for the review @jd-alexander !

About the issue, I think there are two different things:
1. RichText is going to a second line earlier than it should.

This is fixed on https://github.com/WordPress/gutenberg/pull/27275/commits/acdc5546de3c3bd7c947f9aee706def4fe68969a

2. The resize event takes long to happen

This usually happens to me when running metro with Android. Specially when typing, there are many events that takes long to arrive, so the resize event is delayed considerably. This should not happen on the bundled version (without metro).

I have updated and rebased this branch. It's ready to take another look 👀

jd-alexander reviewed (approved):
Thanks @etoledom the button now works as expected on Android. Thanks for confirming the behavior you experience when running the metro server. I experienced the same so it's good to know that's not a bug in any way. LGTM 🚢
